lib/spack/spack/solver/asp.py

```diff
@@ -1839,7 +1839,11 @@ def preferred_variants(self, pkg_name):
 
             # perform validation of the variant and values
             spec = spack.spec.Spec(pkg_name)
-            spec.update_variant_validate(variant_name, values)
+            try:
+                spec.update_variant_validate(variant_name, values)
+            except spack.error.SpackError:
```

Comment:
Do we have a more specific exception for this? Like VariantValueError or whatever?

Reply:
Also, what about requirements?

Reply:
Why guard against a typo `+opnmp` if `+openmp` errors because the majority of the specs don't have that variant. Makes little sense

Reply:
I'll repeat. For requirements `all:` is permissive specific packages are not.

Reply:
See comment below

Reply:
Ok, tried it out, and we have a regression in the logic program :sweat: (so we don't raise here, but emit an unsat problem).